### ISSUE TITLE
Features (not a bug)

### ISSUE THREAD

**Issue by hp6393** [djsudduth/joplin-plugin-paragraph-extractor#2]
Don,

I’ve had a chance to use the plugin on a small project. If you never did another thing to it, this plugin is an excellent tool–exactly what I needed.

Along the line of feature requests for a future version, if you choose to make them:

- There is no option to extract all tags. I have to remember all of the keywords I used and then go through one-by-one and extract them. It would be nice to have the option of having it extract all keywords that are marked with the hashtag indicated (#, @, etc.).

- Add the source document in parenthesis at the end of each extracted paragraph instead of / in addition to as a header. This makes it easier to keep the source with the paragraph as you copy/paste the paragraphs.

- There were times when I wanted to extract a bulleted list as a single paragraph and other times when I wanted to extract each bullet as a separate paragraph. Perhaps this could be controlled by the hash tag? For example, if a bullet is tagged with a # the whole list is extracted as one paragraph and if tagged with a @ the bullet is extracted as its own paragraph.

Again, these are just feature requests/suggestions. Thanks for your work on this project!

**Comment by djsudduth** [comment]
Hi @hp6393 - thank you for the kind comments. Glad you like it!

I have some clarifying questions for each of your bullets:
- When you say "all tags" extracted  - what does that mean as output? Do you want separate extracted paragraph notes for each tag or a single note with all the tags such as "extract any paragraph that has any tag"?
- For trailing source links to the source doc, I assume you want it as body text hyperlink label in parens like (**This is the Source Note**) as a single line after the extracted paragraph?  This should be an easy setting addition.
- You can do the single vs full bullets in settings now but it's either/or for all notes you want extract from - so, are you saying that from one source note you'd want some of the extractions to be single bullets and some to be the full bulleted list?

**Comment by hp6393** [comment]
> Hi @hp6393 - thank you for the kind comments. Glad you like it!
> 
> I have some clarifying questions for each of your bullets:
> 
> * When you say "all tags" extracted  - what does that mean as output? Do you want separate extracted paragraph notes for each tag or a single note with all the tags such as "extract any paragraph that has any tag"?

Yes, I would keep it simple. It could either be one note with headings for each tag or separate notes for each tag, whichever is the easiest to implement.

> * For trailing source links to the source doc, I assume you want it as body text hyperlink label in parens like (**This is the Source Note**) as a single line after the extracted paragraph?  This should be an easy setting addition.

Yes, again something simple to implement. Hyperlinked would be a plus. I was just thinking about putting the {title} in parenthesis at the end of the extracted paragraph. Something like:
> ljkasdlfjk asdljf lkasdj flkja sdlfkj laskjdflkjas dfjahsk dfhkahsdf khasdfkh aksdhfkhas dfjkh akdhsf kajsdhfk hasdkfh kjas hdfk sdahfkh askdj fhkajs dhfkj hasdkfh askjdhf kja hdsfk hasdkf hkahsd f ({note title})

> * You can do the single vs full bullets in settings now but it's either/or for all notes you want extract from - so, are you saying that from one source note you'd want some of the extractions to be single bullets and some to be the full bulleted list?

Yes, where I ran into this was when extracting from several notes. In some notes I wanted the item and in others I wanted the full list. Perhaps this is too much of an edge case.

**Comment by djsudduth** [comment]
@hp6393 - ok thx!  For the last point (bullet 3), would a double hashtag suffice as an override? So, if the setting to pull individual bullets was off/false, a double hashtag could override it. Like this:
------ note 1
- A #mytag
    - B
- C

----- note 2
- DD
- EE ##mytag
- FF

The output would be:
- A #mytag
    - B
- C
- EE ##mytag

Or, maybe in the form of #mytag# instead of ##mytag

**Comment by hp6393** [comment]
> @hp6393 - ok thx! For the last point (bullet 3), would a double hashtag suffice as an override? So, if the setting to pull individual bullets was off/false, a double hashtag could override it. Like this: ------ note 1
> 
> * A #mytag
>   
>   * B
> * C
> 
> ----- note 2
> 
> * DD
> * EE ##mytag
> * FF
> 
> The output would be:
> 
> * A #mytag
>   
>   * B
> * C
> * EE ##mytag
> 
> Or, maybe in the form of #mytag# instead of ##mytag

Yes. To me ## seems more intuitive than #mytag#

**Comment by djsudduth** [comment]
@hp6393  sorry about the delay. I'm adding your second bullet request to add the link back to the source document at the end of each paragraph rather than the top as an option.  Which works better for you of these:

1. Have only the link back to parent note in parentheses (linkback_parent)
2. Have the paragraph number and link back (paragraph# - linkback_parent) 

Is having a paragraph number useful at all?

Example:

In 1917, Adriaan van Maanen discovered van Maanen's Star, an isolated white dwarf. These three white dwarfs, the first discovered, are the so-called classical white dwarfs.  Eventually, many faint white stars were found which had high proper motion, indicating that they could be suspected to be low-luminosity stars close to the Earth, and hence white dwarfs. (<ins>Discovery of White Dwarf Stars</ins>)

Stellar models suggest that the star formed during the collapsing of a molecular cloud and that, after 10 million years, its internal energy generation was derived entirely from nuclear reactions.  (<ins>Sirius A Characteristics</ins>)

**Comment by djsudduth** [comment]
Example:

In 1917, Adriaan van Maanen discovered van Maanen's Star, an isolated white dwarf. These three white dwarfs, the first discovered, are the so-called classical white dwarfs.  Eventually, many faint white stars were found which had high proper motion, indicating that they could be suspected to be low-luminosity stars close to the Earth, and hence white dwarfs. (<ins>Discovery of White Dwarf Stars</ins>)

Stellar models suggest that the star formed during the collapsing of a molecular cloud and that, after 10 million years, its internal energy generation was derived entirely from nuclear reactions.  (<ins>Sirius A Characteristics</ins>)

**Comment by hp6393** [comment]
No problem. We've all got "day jobs!"

Re your question, either one is helpful. Having the paragraph number is probably the best. It helps quickly locate the source.
________________________________

**Comment by djsudduth** [comment]
@hp6393 ~~Still debating here~~ - the paragraph number is somewhat confusing since headers are considered paragraphs and then bullets can be subdivided - so that would be a bullet count within a paragraph.

I'm not going to do it - looks weird and is confusing to have the numbers

**Comment by hp6393** [comment]
Yes, I can see how that could be confusing / misinterpreted.

**Comment by djsudduth** [comment]
Option to have backlinks at the end of each paragraph had been added now in Version 1.1.5. Tested and released.

- `Embed source note titles at end of each extracted paragraph`: Titles of source notes will be embedded at the end of each extracted paragraph rather than at the top with a link back to the original source notes. Default `false`.

**Comment by hp6393** [comment]
Works really well. Nice addition!

**Comment by djsudduth** [comment]
@hp6393  - are you still using this plugin?

**Comment by hp6393** [comment]
I do, Don, but recently have had some health issues in my family, so I haven't used it in awhile. I do think it is a very useful plug-in.
________________________________

**Comment by djsudduth** [comment]
@hp6393 - Happy New Year! I'm just about ready to add a new feature that will refresh extracted notes from their sources rather than creating a new note. It's not automated with a timer yet and can only refresh one note at time. But, it allows you to alter the source notes and go back to an extracted note to refresh with any changes made to the original notes.

That of interest to you?  Hope your health is better.

**Comment by djsudduth** [comment]
Closing this issue for now. If some issues come up, it will be added as new.

**Comment by hp6393** [comment]
Don, I just saw your email. It has been a turbulent year health wise (family in ICU, transplant, etc.) but starting to get back into the swing of things. I appreciate your continued work on this plug-in.


________________________________
